src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpRequestAdapter.java

```diff
@@ -117,6 +117,14 @@ public String getHeaderValue(String headerName) {
     return this.headers.getAll(headerName);
   }
 
+  @Override
+  public MultiMap<String, String> getHeaders() {
+    if (this.headers == null) {
+      initializeHeaders();
+    }
+    return this.headers;
```

Comment:
Is this immutable? Should be really shouldn't it, but API isn't clear about this.

Reply:
it is, the call to `initializeHeaders` sets it as immutable.